common/src/api/external/error.rs

```diff
@@ -151,6 +151,14 @@ impl Error {
     pub fn unavail(message: &str) -> Error {
         Error::ServiceUnavailable { internal_message: message.to_owned() }
     }
+
+    /**
+     * Generates an [`Error::Unauthenticated`] error with the specific
+     * message
+     */
+    pub fn unauthenticated(message: &str) -> Error {
```

Comment:
This might be fine but it makes me a little nervous.  Take a look at this conversion from the various authn errors to an *HttpError* (*not* this `Error`):
https://github.com/oxidecomputer/omicron/blob/a5713c63994feb95043cf842931048696889c479/nexus/src/authn/mod.rs#L217-L246

There are a few considerations there (admittedly that we've mostly punted on so far), and some of them are critical for security.  I think we want to handle this all in one place.  So I'm afraid about making it easy to add new places where we generate `Unauthenticated` and potentially get this wrong.

I wonder if we can achieve the same thing by just doing an authz check.  If the user is authenticated, they'll presumably not be able to do anything, and the authz system has to be trusted to produce the right kind of error.

Reply:
Handled with `actor_required()` as suggested in chat.